js/services/component.ts

```diff
@@ -334,3 +337,27 @@ export const onOutcome = (outcome: any, objectData: any[], flowKey: string): JQu
             }
         });
 };
+
+export const getPageSize = (model, flowKey) => {
+
+    const pageLimitFromAttributes = pathOr(null, ['attributes', 'paginationSize'], model);
+
+    const usePaginationAttribute = Utils.isEqual(model.attributes.pagination, 'true', true)
+        && !Number.isNaN(pageLimitFromAttributes);
+
+    const pageLimitSettingForComponentType = Settings.flow(
+        `paging.${model.componentType.toLowerCase()}`, flowKey,
+    );
+
+    const pageLimitSettingFromListFilter = pathOr(null, ['objectDataRequest', 'listFilter', 'limit'], model);
+
+    const limit = usePaginationAttribute
+        ? pageLimitFromAttributes // 1st priority
+        : (
+            pageLimitSettingForComponentType // 2nd priority
+            || pageLimitSettingFromListFilter // 3rd priority
```

Comment:
I have a feeling 2nd and 3rd priority should be swapped here, and then we wouldn't have to get rid of the table default from the settings.
What do you think?
Currently with this, users can still run into the same problem in the ticket for all the other components.

Reply:
Yes, that's what this statement in the issue does suggest: "The UI shouldn't be substituting its own value here (or even defaulting to anything, unless there's no listFilter at all)". I'll switch those around.